### PR TITLE
Allow custom losses to have mismatching target shapes.

### PR DESCRIPTION
**Pull request by hgaiser**
In response to [these](https://github.com/fchollet/keras/pull/8056#pullrequestreview-76745344) comments, this PR partly reverts the changes made in https://github.com/fchollet/keras/pull/8056 . That PR had the unintended side effect of forcing target blobs to have the same shape as output blobs for custom loss functions, which is currently frequently used as a workaround ([discussion](https://github.com/fchollet/keras/issues/7395)).

### CODE REVIEW

**Comment by hgaiser** [comment]
Thank you. Do you have any idea if a v2.1.2 is to be released soon? This bug was preventing me from using 2.1.0+.
